src/components/ComponentPages/News/Edit/index.tsx

```diff
@@ -1,154 +1,139 @@
-import { Form, Input, Button, Checkbox, Divider, Card, Badge } from "antd";
+import React, { useState, useEffect } from "react";
+import {
+  Form,
+  Button,
+  Checkbox,
+  Spin,
+  Typography,
+  Input,
+  Row,
+  Col,
+  Card,
+} from "antd";
 import { useRouter } from "next/router";
-import { Row, Col } from "antd";
-import React, { useState } from "react";
-import { updateNewsDataApi } from "../../../../network/api/campNewsApi";
-import styles from "../addEditNews.module.scss";
-import { Spin, Typography } from "antd";
 import { LoadingOutlined } from "@ant-design/icons";
-import { RootState } from "../../../../store";
-import { useSelector } from "react-redux";
+import "antd/dist/antd.css";
+import { updateNewsFeedApi } from "../../../../network/api/campNewsApi";
+import { getEditCampNewsFeedApi } from "../../../../network/api/campNewsApi";
+import useAuthentication from "../../../../../src/hooks/isUserAuthenticated";
+import styles from "../addEditNews.module.scss";
 
-const { Text } = Typography;
 const antIcon = <LoadingOutlined spin />;
+const { Text } = Typography;
 
 export default function Edit() {
-  const dataToUpdate = useSelector(
-    (state: RootState) => state?.campNews?.campNews?.newsToEdit
-  );
-
-  const [urlErrorMsg, setUrlErrorMsg] = useState("");
-  const [urlError, setUrlError] = useState(
-    new Array(dataToUpdate.length).fill(false)
-  );
+  const [dataToUpdate, setDataToUpdate] = useState({});
   const [loading, setLoading] = useState(false);
+  const [urlErrorMsg, setUrlErrorMsg] = useState("");
+  const [urlError, setUrlError] = useState(false);
   const router = useRouter();
+  const isLogin = useAuthentication();
   const [form] = Form.useForm();
+
   const goBack = () => {
     setLoading(true);
     router.back();
   };
+
   const onFinish = async (values: any) => {
     setLoading(true);
-    const dataObj = {
-      topic_num: +router.query?.camp[0]?.split("-")[0],
-      camp_num: +router.query?.camp[1]?.split("-")[0],
-      id: values.data.map((id) => id.id),
-      display_text: values.data.map((text) => text.display_text),
-      link: values.data.map((link) => link.link),
-      available_for_child: values.data.map(
-        (available) => available.available_for_child
-      ),
-    };
-    const res = await updateNewsDataApi(dataObj);
+    const res = await updateNewsFeedApi({
+      newsfeed_id: dataToUpdate?.id,
```

Comment:
Showing an error `Property 'id' does not exist on type`

Reply:
fixed